Compiler bug with wrong argument number in comp time

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.04
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1) 10.3.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-01-09 13:59:41
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 c1711b8.22fce6a

Git version: git version 2.30.2
Git vroot status: weekly.2022.01-11-g22fce6a5
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/pattern_framwork/0sample_bug.v`
```v
module main

struct S1 {}

fn  main() {

	s1 := S1{}

	$for method in S1.methods {
		println(s1.$method('yo'))
	}
}

fn (t S1) method_hello() string {
	return 'Hello'
}

```

**What did you expect to see?**

A formated error explaining the number of argument is wrong


**What did you see instead?**
```
==================
/tmp/v_1000/0sample_bug.9936956607750145413.tmp.c:10950: error: too few arguments to function
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/tests/comptime_call_method_args_err.out
+++ b/vlib/v/checker/tests/comptime_call_method_args_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/comptime_call_method_args_err.vv:9:14: cgen error: expected 0 arguments to method S1.method_hello, but got 1
+    7 |
+    8 |     $for method in S1.methods {
+    9 |         println(s1.$method('yo'))
+      |                    ~~~~~~~~~~~~~
+   10 |     }
+   11 | }

--- a/vlib/v/checker/tests/comptime_call_method_args_err.vv
+++ b/vlib/v/checker/tests/comptime_call_method_args_err.vv
@@ -1,0 +1,15 @@
+module main
+
+struct S1 {}
+
+fn main() {
+	s1 := S1{}
+
+	$for method in S1.methods {
+		println(s1.$method('yo'))
+	}
+}
+
+fn (t S1) method_hello() string {
+	return 'Hello'
+}

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -99,12 +99,17 @@ fn (mut g Gen) comptime_call(mut node ast.ComptimeCall) {
 		}
 		// check argument length and types
 		if m.params.len - 1 != node.args.len && !expand_strs {
-			// do not generate anything if the argument lengths don't match
-			g.writeln('/* skipping ${sym.name}.$m.name due to mismatched arguments list */')
-			// g.writeln('println(_SLIT("skipping ${node.sym.name}.$m.name due to mismatched arguments list"));')
-			// eprintln('info: skipping ${node.sym.name}.$m.name due to mismatched arguments list\n' +
-			//'method.params: $m.params, args: $node.args\n\n')
-			// verror('expected ${m.params.len-1} arguments to method ${node.sym.name}.$m.name, but got $node.args.len')
+			if g.inside_call {
+				g.error('expected ${m.params.len - 1} arguments to method ${sym.name}.$m.name, but got $node.args.len',
+					node.pos)
+			} else {
+				// do not generate anything if the argument lengths don't match
+				g.writeln('/* skipping ${sym.name}.$m.name due to mismatched arguments list */')
+				// g.writeln('println(_SLIT("skipping ${node.sym.name}.$m.name due to mismatched arguments list"));')
+				// eprintln('info: skipping ${node.sym.name}.$m.name due to mismatched arguments list\n' +
+				//'method.params: $m.params, args: $node.args\n\n')
+				// verror('expected ${m.params.len-1} arguments to method ${node.sym.name}.$m.name, but got $node.args.len')
+			}
 			return
 		}
 		// TODO: check argument types

